packages/utils/src/metrics.ts

```diff
@@ -0,0 +1,80 @@
+export type LabelsGeneric = Record<string, string | number>;
+export type LabelKeys<Labels extends LabelsGeneric> = Extract<keyof Labels, string>;
+export type CollectFn<Labels extends LabelsGeneric> = (metric: Gauge<Labels>) => void;
+
+// Type `{}` is required here as alternative types do not work
+// eslint-disable-next-line @typescript-eslint/ban-types
+export type NoLabels = {};
```

Comment:
What you actually want to use here is 

```suggestion
export type NoLabels = Record<string, never>;
```

Try this script; 

```ts
interface Gauge<Labels extends LabelsGeneric = NoLabels> {
    labels: Labels;
}

// Though we specified this gauge to have no labels 
// but it will not raise type error for `{}` type. 
const g1: Gauge = {
    labels: {
        a: 12,
        b: "string value"
    }
}
```

Reply:
As mentioned in the comment there were some issues related to using another type than `{}`, the example you provided is not how the Gauge type looks that I added in this PR, it's not possible to add labels unless the type defines it.

Previous type error
```
@lodestar/validator:   Types of property 'dbApproximateSizeTime' are incompatible.
@lodestar/validator:     Type 'Histogram<LabelsGeneric>' is not assignable to type 'Histogram<NoLabels>'.
@lodestar/validator:       Type 'LabelsGeneric' is not assignable to type 'NoLabels'.
@lodestar/validator:         'string' index signatures are incompatible.
@lodestar/validator:           Type 'string | number' is not assignable to type 'never'.
@lodestar/validator:             Type 'string' is not assignable to type 'never'.
@lodestar/validator: 132         this.db.setMetrics(metrics.db);
```

However, it looks like this issue gets resolve if I do `<Labels extends LabelsGeneric = NoLabels>` everywhere, I kinda wanted to avoid that and as it is better to either have to explicitly set the label type or have it inferred from the provided config but this works as well. The main issue with setting a default type is that you might forget to properly forward the generic type to subtypes.

see commit https://github.com/ChainSafe/lodestar/pull/6201/commits/2a14a4b0e2f22d03138ebae8fbc62ce5f991a79e